python/mp/triggers/trigger.py

```diff
@@ -0,0 +1,156 @@
+import abc
+import copy
+
+from mp import color
+
+
+class Trigger(abc.ABC):
+    """
+    Trigger is the base class for all triggers.
+
+    Triggers by their nature have less customizing than effects.
+    You're really only expected to define the effect_sequence in any subclass.
+
+    * effect_sequence[]: a list of dictionary objects defining what effects
+                         should 
+    """
+
+    def __init__(self, name):
+
+        # Informs the OSC path naming, otherwise not very useful here
+        self.name = name
+
+        # Technically used to compare one trigger to another
+        self.id = -1
+
+        # Used to indirectly access effects
+        self.rosary = None
+
+        # The trigger() method will reset this to 0 every time we
+        # go through the sequence
+        self.time = 0
+
+        # This should follow self.time until we stop triggering
+        self.last_trigger = 0
+
+        # How many ticks are allowed to pass between last_trigger and time
+        # before we decide we've lost the signal
+        self.fade_out_threshold = 30.0
+
+        # How many "ticks" between losing the trigger and killng the effects
+        self.fade_out_duration = 30.0
+
+        # Effects come and go, but a trigger is forever. FOR EV ER.
+        # The only difference is if we're currently running or not
+        self.running = False
+
+        # A list of effects
+        self.effect_sequence = self.set_effect_sequence()
+
+
+    def __eq__(self, other):
+        return (self.id == other)
+
+    def __repr__(self):
+        return "<Trigger:{}: id={}>".format(self.name, self.id)
+
+    def get_name(self):
+        """Returns the name of the Trigger. (Does anybody care?)"""
+        return self.name
+
+    @abc.abstractmethod
+    def set_effect_sequence(self):
+        pass
```

Comment:
I could've just said "hey people, set `self.effect_sequence` in all your `__init__`s", but I needed an abstract method in here somewhere for `inspect.isabstract` to recognize this class as an abstract class.